kedro/config/omegaconf_config.py

```diff
@@ -421,15 +430,24 @@ def _register_new_resolvers(resolvers: dict[str, Callable]) -> None:
                 _config_logger.debug(msg)
                 OmegaConf.register_new_resolver(name=name, resolver=resolver)
 
-    @staticmethod
-    def _check_duplicates(seen_files_to_keys: dict[Path, set[Any]]) -> None:
+    def _check_duplicates(self, key: str, config_per_file: dict[Path, Any]) -> None:
+        if key == "parameters":
+            seen_file_to_keys = {
+                file: self._get_all_keys(OmegaConf.to_container(config, resolve=False))
```

Comment:
isn't `config` already a `DictConfig`? why do you need to convert it again?

Reply:
So, DictConfig `.items()` method resolves the values and that's not what I want. So, I convert the DictConfig to an unresolved dictionary. Let me know if you have some alternative/better way to handle this.

Reply:
Can you give me an example with just DictConfig? I am not sure if I am following, if config is resolved here already how would converting it back to DictConfig helps?

Reply:
See the line 361 in this module. We are doing an iteration over the dictionary keys and values. Copying the function here:
```python
    def _get_all_keys(self, cfg: Any, parent_key: str = "") -> set[str]:
        keys: set[str] = set()

        for key, value in cfg.items():
            full_key = f"{parent_key}.{key}" if parent_key else key
            if isinstance(value, dict):
                keys.update(self._get_all_keys(value, full_key))
            else:
                keys.add(full_key)
        return keys
```
With `cfg` being `DictConfig`, the iteration would resolve the interpolations, e.g, `${test_env}` in one of our tests. But if I do `OmegaConf.to_container(cfg, resolve=False)` it will return a python `dict` and not `DictConfig` and since I use `resolve=False` it won't resolve the interpolations.

Reply:
Thanks, this make sense!